src/components/Personalization/in-app-message-actions/actions/displayIframeContent.js

```diff
@@ -148,60 +87,145 @@ export const createIframe = (htmlContent, clickHandler) => {
   return element;
 };
 
-export const createContainerElement = settings => {
-  const { mobileParameters = {}, webParameters = {} } = settings;
-  const element = document.createElement("div");
-  element.id = ELEMENT_TAG_ID;
-  element.className = `${ELEMENT_TAG_CLASSNAME}`;
-  Object.assign(
-    element.style,
-    buildStyleFromParameters(mobileParameters, webParameters)
-  );
+const displayBasedOnWebParams = (iframe, webParameters, container, overlay) => {
+  const { style: iframeStyle = {} } = webParameters[ALLOY_IFRAME_ID];
+  const {
+    style: messagingStyle = {},
+    params: messagingParams = {}
+  } = webParameters[ALLOY_MESSAGING_CONTAINER_ID];
 
-  return element;
+  const {
+    style: overlayStyle = {},
+    params: overlayParams = {}
+  } = webParameters[ALLOY_OVERLAY_CONTAINER_ID];
+
+  assign(iframe.style, iframeStyle);
+  assign(container.style, messagingStyle);
+
+  if (overlayParams.enabled) {
```

Comment:
use object destructuring for this too.  

`const {enabled=false} = overlayParams;`

Reply:
i'd like to see this code be more generic.

ideally you could have code that:

1. iterates across Object.keys(webParameters)
2. checks of one of the keys matches the id of an element created.
3. pass the element and the `{style:{}, params:{}}` object to a `renderElement` function.
4. The renderElement function applies the css from `style` to the element passed in.  It also looks for known params and applies them to the element as well.  note that this function would be used for both the messaging container and the overlay.

Reply:
Using object destructuring allows you to set defaults for params.  We need to do that or validate the `webParameters` object.

Reply:
You've made a lot of great improvements here.  But you can take it even further to simplify the use of `webParameters` the way I described above.  To illustrate, take a look at this code https://github.com/adobe/alloy/pull/1039

Reply:
done